test/extensions/transport_sockets/tls/integration/ssl_integration_test.cc

```diff
@@ -399,7 +399,7 @@ name: "envoy.tls.cert_validator.timed_cert_validator"
   const auto* socket = dynamic_cast<const Extensions::TransportSockets::Tls::SslHandshakerImpl*>(
       connection->ssl().get());
   ASSERT(socket);
-  while (socket->state() != Ssl::SocketState::HandshakeInProgress) {
+  while (socket->state() < Ssl::SocketState::HandshakeInProgress) {
```

Comment:
Hi, I think it's confusing to use `<` for enum type, maybe we can replace it with `== Ssl::SocketState::PreHandshake`.

Reply:
```suggestion
  while (socket->state() == Ssl::SocketState::PreHandshake) {
```